data-raw/02_process_mars_gribs_to_COGS.py

```diff
@@ -0,0 +1,88 @@
+import os
+import tempfile
+from pathlib import Path
+
+import pandas as pd
+import xarray as xr
+
+from src.blob_utils import (
+    download_file,
+    list_blobs,
+    load_env_vars,
+    upload_file,
+)
+
+SAS_TOKEN, CONTAINER_NAME, STORAGE_ACCOUNT = load_env_vars()
+
+grib_files = list_blobs(
+    sas_token=SAS_TOKEN,
+    container_name=CONTAINER_NAME,
+    storage_account=STORAGE_ACCOUNT,
+    name_starts_with="ds-aa-afg-drought/raw/",
+)
+
+grib_files = [file for file in grib_files if file.endswith(".grib")]
+
+for blob_name in grib_files:
+    print(f"name: {os.path.basename(blob_name)}")
+
+    # Create a temporary file
+    with tempfile.TemporaryDirectory() as td:
+        temp_base_path = os.path.basename(blob_name)
+        td = Path(td)
+        tf = td / temp_base_path
+        # Download the file
+        download_file(
+            sas_token=SAS_TOKEN,
+            container_name=CONTAINER_NAME,
+            storage_account=STORAGE_ACCOUNT,
+            blob_path=blob_name,
+            local_file_path=tf,
+        )
+
+        # Open the file with xarray
+        ds = xr.open_dataset(
+            tf,
+            engine="cfgrib",
+            drop_variables=["surface", "values"],
+            backend_kwargs=dict(time_dims=("time", "forecastMonth")),
+        )
+
+        ds["longitude"] = (ds["longitude"] + 180) % 360 - 180
+        ds["precip"] = (
+            ds["tprate"] * ds["time"].dt.days_in_month * 24 * 3600 * 1000
+        )
+        ds = ds.drop_vars("tprate")
+        da = ds["precip"]
+        da_mean = da.mean(dim=["number"])
+
+        # loop through pub dates & COGS
+        pub_dates = da_mean.time.values
+        forecast_months = da_mean.forecastMonth.values
+
+        # give a CRS
+        da_mean = da_mean.rio.write_crs("EPSG:4326", inplace=False)
+
+        for pub_date in pub_dates:
+            da_out = da_mean.sel(time=pub_date)
+            date_str = pd.to_datetime(pub_date).strftime("%Y-%m-%d")
+            filename_prefix = f"seas5_mars_tprate_em_{date_str}"
+            print(filename_prefix)
+
+            fms = da_out.forecastMonth.values
+            for i, fm_tmp in enumerate(fms):
```

Comment:
can just do `for fm_tmp in fms:`?

Reply:
good call! I think the `enumerate` process was left over from different code where I was merging multiple bands into one tif and using the `i` to specify the band position... but yea no necessary here any more as leadtimes are getting split by file. Addressed here: [0674cdb](https://github.com/OCHA-DAP/ds-aa-afg-drought/pull/4/commits/0674cdba1490a3e4708c4c3a92ddf1ac392890f0)